Sources/NIO/EventLoopFuture.swift

```diff
@@ -197,6 +197,25 @@ public struct EventLoopPromise<Value> {
         future.cascade(to: self)
     }
 
+    /// Complete the promise with the passed in `Result<Value, Error>`.
+    ///
+    /// This method is equivalent to invoking:
+    /// ```
+    /// switch result {
+    /// case .success(let value):
+    ///     promise.succeed(value)
+    /// case .failure(let error):
+    ///     promise.fail(error)
+    /// }
+    /// ```
+    ///
+    /// - parameters:
+    ///     - result: The result which will be used to succeed or fail this promise.
+    @inlinable
+    public func completeWith(_ result: Result<Value, Error>) {
```

Comment:
Not super happy about the overloading of `completeWith` but didn't want to add an explicit `result` label.

Reply:
let's CC @Lukasa  & @normanmaurer for the naming. Could also always do `public func fulfill(_ result: Result)`?

Reply:
I think `fulfill(_ result: Result)` is nice, gives you a nice and readable way to pass a promise "as a completion handler" for callback based APIs

```
func doSomething(completion: @escaping (Result<String, Error>) -> ()) {
  ...
  completion(.success("Done"))
}

let p: EventLoopPromise<String> = ...

doSomething(completion: p.fulfill)

p.futureResult.whenComplete { ... }
```

Reply:
So far I've not yet seen how such kind of overload could end up being problematic... as it reads quite well (and going for another name like `fulfill` or other, seems a bit ad hoc if only to avoid an overload) I'm not sure if avoiding the overload is worth it here... Any secret swift knowledge I'm missing about such specific overload is bad?

(more complex ones, esp. with differing return types I'd be on board with avoiding, here it seems to help more than hurt I thought)

Reply:
I'm actually fine with `completeWith(_ result: Result)` but I know not everyone is a fan of overloads which is why I called it out.

I'm 👎 on `fulfil` for the consistency reason.

Reply:
I don't like overloads but in this particular case I'm probably fine with it.